.github/workflows/build_nuget.yml

```diff
@@ -25,12 +25,13 @@ jobs:
         with:
           submodules: 'recursive'
       - uses: ilammy/msvc-dev-cmd@v1
+      # NASM required to compile OpenSSL assembly language code
+      - uses: ilammy/setup-nasm@v1
       - name: Dependencies
         run: >
           vcpkg install
           --host-triplet=${{ matrix.arch_config.arch }}-windows
           --triplet=${{ matrix.arch_config.arch }}-windows
-          openssl-windows
```

Comment:
Could you explain why we cant use the vcpkg provided instance of openssl?

Reply:
The vcpkg one is dynamically linked and Griffin found that the resulting package won't work on a machine without openssl already installed.

Reply:
Apparently Chad wants absolutely no DLLs in the nuget package. I could try using the -static vcpkg and including that library in our package. But at this point we already have the build working for a vendored openssl, and compiling it ourselves gives us control of what flags the library gets built with.

What would be the motivation for using a prepackaged openssl over compiling it ourselves?

Reply:
The advantage of consuming vcpkg is reduced friction when we need to upgrade version. But I can understand the desire to make it work and have no deps

Reply:
I am a little concerned about the ongoing maintenance burden of doing a from source vendored build of openssl

Reply:
I would expect a "git pull" in the submodule to be enough to update version. The build system shouldn't change.

There was an issue we recently found where the default static build will always create a separate PDB file with debug symbols. If using vcpkg static openssl, we either must tell the end user of the nuget package to install static openssl with vcpkg, or we have to copy the PDB into the nuget package and set up the right path to it. If the PDB file is missing, you get a ton of linker warnings when linking a program with rlclientlib.

Having a vendored openssl and compiling it ourselves lets us modify the compiler flags to omit references to the PDB file. I think this is a good reason to go with vendored openssl.

Reply:
Okay, sounds good. Lets go with vendored